src/Domain/Insights/Syntax.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\PhpInsights\Domain\Insights;
+
+use NunoMaduro\PhpInsights\Domain\Contracts\GlobalInsight;
+use NunoMaduro\PhpInsights\Domain\Contracts\HasDetails;
+use NunoMaduro\PhpInsights\Domain\Details;
+use PHP_CodeSniffer\Config;
+use Symfony\Component\Process\Process;
+
+/**
+ * @internal
+ */
+final class Syntax extends Insight implements HasDetails, GlobalInsight
+{
+    /**
+     * @var array<Details>
+     */
+    private $details = [];
+
+    public function getTitle(): string
+    {
+        return 'Syntax Check';
+    }
+
+    public function hasIssue(): bool
+    {
+        return count($this->details) > 0;
+    }
+
+    public function getDetails(): array
+    {
+        return $this->details;
+    }
+
+    public function process(): void
+    {
+        $phpPath = (string) Config::getExecutablePath('php');
+        $filesToAnalyse = array_map(static function (string $file): string {
+            return escapeshellarg($file);
+        }, $this->filterFilesWithoutExcluded($this->collector->getFiles()));
+
+        $cmdLine = sprintf(
+            '%s %s --no-colors --no-progress --json %s',
+            escapeshellcmd($phpPath),
+            escapeshellarg(getcwd() . '/vendor/bin/parallel-lint'),
```

Comment:
Can we use `__DIR__ ` instead of `getcwd`, as `getcwd` depends on where we started the program.

Reply:
You are totally right !